lib/lseek64_stubs.c

```diff
@@ -46,12 +46,10 @@ CAMLprim value stub_lseek64_data(value fd, value ofs) {
   off_t c_ret;
 
   caml_enter_blocking_section();
-#if defined(SEEK_DATA)
   c_ret = lseek(c_fd, c_ofs, SEEK_DATA);
-#else
-  /* Set the file pointer to ofs; pretend there is data */
-  c_ret = lseek(c_fd, c_ofs, SEEK_SET);
-#endif
+  /* If there is no SEEK_DATA support, 
+     set the file pointer to ofs; pretend there is data */
+  if (c_ret == -1) c_ret = lseek(c_fd, c_ofs, SEEK_SET);
```

Comment:
You should only ever fall back in the specific case of SEEK_HOLE not being available.

Errors such as EBADF or ESPIPE shouldn't result in a fallback.